src/editors/data/services/cms/urls.js

```diff
@@ -7,26 +7,34 @@ export const unit = ({ studioEndpointUrl, unitUrl }) => (
 );
 
 export const returnUrl = ({ studioEndpointUrl, unitUrl, learningContextId }) => {
-  if (learningContextId && learningContextId.includes('library-v1')) {
+  if (learningContextId && learningContextId.startsWith('library-v1')) {
     // when the learning context is a v1 library, return to the library page
     return libraryV1({ studioEndpointUrl, learningContextId });
   }
+  if (learningContextId && learningContextId.startsWith('lib')) {
+    // when it's a v2 library, there will be no return url (instead a closed popup)
+    return '';
+  }
   // when the learning context is a course, return to the unit page
   return unitUrl ? unit({ studioEndpointUrl, unitUrl }) : '';
 };
 
 export const block = ({ studioEndpointUrl, blockId }) => (
-  blockId.includes('block-v1')
+  blockId.startsWith('block-v1')
     ? `${studioEndpointUrl}/xblock/${blockId}`
-    : `${studioEndpointUrl}/api/xblock/v2/xblocks/${blockId}`
+    : `${studioEndpointUrl}/api/xblock/v2/xblocks/${blockId}/fields/`
 );
 
 export const blockAncestor = ({ studioEndpointUrl, blockId }) => (
-  `${block({ studioEndpointUrl, blockId })}?fields=ancestorInfo`
+  blockId.startsWith('block-v1')
+    ? `${block({ studioEndpointUrl, blockId })}?fields=ancestorInfo`
+    : '' // this url only need to get info to build the return url, which isn't used by V2 blocks
```

Comment:
Shouldn't we instead just not make this call in that case?

Reply:
<shrug> maybe, but it also makes sense to check the blockId here, until we refactor all this after moving everything to popups.

Reply:
specifically: wouldn't this lead to an error, as it would attempt to make a an axios call at ''?

Reply:
or maybe we want this to error loudly if it is used, and instead of returning '' we should raise an error.